### DISCUSSION TITLE
Plugin for block ip?

### DISCUSSION THREAD

**Discussion by itiloo** [rejetto/hfs#578] [Q&A]
Hi Rejetto:

do you have any plugin about block ip after x number of error logins?.

Thanks.

**Comment by rejetto** [comment]
yes, it's installed and enabled by default.
it's called "antibrute"

**Comment by itiloo** [comment]
Yes, but that plugin not block ip, true?

**Comment by rejetto** [comment]
right, it doesn't. 
would it be good?

**Comment by itiloo** [comment]
How anti-ddos plugin but not with request, with logins.

**Comment by rejetto** [comment]
seems legit, i will consider this for the near future, like in 0.53 .
nice thing is that, being it a plugin, a programmer could make it without waiting for me.

**Reply by PH5HP** [reply]
Hi,

I would like this too ... I am DDOS attacked on a daily basis ....

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

**Reply by rejetto** [reply]
a DDOS is when a single attacker uses multiple computers with the purpose to block your service, AFAIK.
maybe your case is different? if you show a bit of your suspicious log lines I may suggest something.

**Reply by PH5HP** [reply]
Hi Rejetto,

Maybe not DDOS, but someone from China tries something almost every day. Screenshot below.

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

![image](https://github.com/rejetto/hfs/assets/140949730/c972e791-8581-4640-b456-24b79055501e)

**Reply by rejetto** [reply]
that's why i asked.
look, here we are discussing about login attempts, and your screenshot shows none.
when i'll add the feature discussed above, these requests won't be blocked (unless they are not preceeded by login attempts).
you have first to decide what is the rule you want to apply. Can't you just block that ip?

**Reply by PH5HP** [reply]
Hi,

I understand. I had forgotten this was all about log in attemps. Sorry.I thought it was also about this kind of misbehaviour from a user. What is he trying to do or what is he doing to generate this messages? This happens on a daily basis. Only from China.

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

**Reply by rejetto** [reply]
he is making http requests with some software that's not a browser.
its purpose, i don't know. I can't infer anything from your screenshot, but i also cannot see anything harmful.
http is not only for browser, you can make requests even from the command line.
did you block that ip?

**Reply by PH5HP** [reply]
Hi Rejetto,

TNX for the swift reply. Not yet, but I am going to. :-)

Kind regards,
Henk - PH5HP - fileserver.ph5hp.nl

**Comment by itiloo** [comment]
Ok, perfect, thank you so much

**Comment by rejetto** [comment]
you'll find this in latest version 
https://github.com/rejetto/hfs/releases/tag/v0.53.0-alpha1

**Reply by oromanobr** [reply]
hi. I accidentally deleted the anti-brute plugin. How do I reinstall it?

**Reply by rejetto** [reply]
@oromanobr  create the folder you deleted, and download only this file in it https://github.com/rejetto/hfs/blob/main/plugins/antibrute/plugin.js

**Comment by ZIKVLD** [comment]
![image](https://github.com/rejetto/hfs/assets/154645533/9427a31a-9fe3-4af7-8213-211593925a87)
+
Plugins broken:
download-counter 5.11
file-icons by rejetto 2.3
thumbnails by rejetto 3.12
Perhaps something else

**Reply by rejetto** [reply]
thanks for reporting, and sorry for that.
i'm removing this version

**Comment by itiloo** [comment]
> you'll find this in latest version https://github.com/rejetto/hfs/releases/tag/v0.53.0-alpha1

Thank you so much Rejetto for your work.

I have tested the plugin and it blocks the IP correctly after the specified attempts. I have 2 doubts.

1.-Is each attempt for that IP cumulative daily and at the end of the day is that counter reset? Or is the attempt counter not cleared at any time and continue accumulating for life until it reaches that specified number? .
2.-To manually unblock the IP, is it necessary to go to config.yaml and delete the corresponding line?

Thank you very much Rejetto.

**Reply by rejetto** [reply]
1.  counter is reset after successful login, but also after 10x the "max delay". This latter part was ok when there was no "block" feature, but now i'm not sure. But it's good need anyway to have a time when I discard the information, otherwise if you get attempts from many IPs, you'll have some memory wasted forever (until restart). I'm not sure what's best here, maybe reset after 24 hours?
2. edit config.yaml is ok, but you have blocked IP in Admin-panel/Options.

**Reply by itiloo** [reply]
1.-I think (in my opinion) that at the end of the day, for example, that counter of the accumulated failed attempts that that IP has accumulated is reset at the end of the day, or another option would be to add a time condition, for example, blocking the ip if it makes x attempts in the space of x seconds.
2.-Ok, thanks Rejetto, I couldn't find where it was in the administration panel.

**Reply by rejetto** [reply]
1. i may go for the 24 hours. Better not introduce options if not useful.
2. maybe a link to it would have been useful

**Reply by itiloo** [reply]
I see better option block ip if it makes x failed attempts in the time space of x seconds, but you decide.

**Reply by rejetto** [reply]
you need something different than 24 hours?

**Reply by itiloo** [reply]
I understand that whoever wants to try to access by brute force will automate, for example, 5 attempts every 20 seconds, so it could be good to have the option to block IP if, for example, 5 failed attempts are made in the space of 20 seconds.

**Reply by rejetto** [reply]
you are forgetting that this plugin is delaying you, so you cannot do 5 attempts in 20 seconds.
with default settings it will take 50 seconds for 5 attempts, even automated.

**Reply by itiloo** [reply]
True, sorry.
